tests/test_parser.py

```
@@ -204,3 +205,73 @@ def test_malformed_config_reported(self, scenario_report: str):
+        test_dir = tmp_path / "tests"
+        test_dir.mkdir()
+        dse_toml = test_dir / "dse_qwen_30b_a3b.toml"
+        single_toml = test_dir / "qwen_30b_a3b.toml"
+        self._write_test_toml(dse_toml, "dse_qwen_30b_a3b")
+        self._write_test_toml(single_toml, "dse_qwen_30b_a3b")
+
+        parser = TestParser([dse_toml, single_toml], None)  # type: ignore
+        fake = self._make_fake_parsed("dse_qwen_30b_a3b")
+
+        with patch.object(parser, "_parse_data", return_value=fake):
+            with pytest.raises(ValueError, match="dse_qwen_30b_a3b"):
```

Ruff (0.15.10)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)